ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/Common.hs

```diff
@@ -65,6 +68,21 @@ takePrefix :: PrefixLen -> BL.ByteString -> ShortByteString
 takePrefix (PrefixLen n) =
     Short.toShort . BL.toStrict . BL.take (fromIntegral n)
 
+{-------------------------------------------------------------------------------
+  Pruning
+-------------------------------------------------------------------------------}
+
+-- | The "tip" to prune snapshots from.
+--
+-- `SecurityParam` has been updated to use `NonZero` but we need to prune from
+-- @0@ in some cases.
+--
+-- Rather than using a plain `Word64` we use this to be able to distinguish that
+-- we are indeed using
+--   1. @0@ in places where it is necessary
+--   2. the security parameter as is, in other places
+data LedgerDBPruneTip = LedgerDBPruneTipZero | LedgerDBPruneTip SecurityParam
```

Comment:
This should probably be in the LedgerDB folder. Also some rewording:
```suggestion
-- | How many ledger states to keep in the LedgerDB after a prune.
--
-- Rather than using a plain `Word64` we use this to be able to distinguish that
-- we are indeed using
--   1. @0@ in places where it is necessary, namely when completing the replay of the ImmutableDB blocks as part of opening the ChainDB
--   2. the security parameter as is, in other places, note it is a strictly positive value
data LedgerDBPruneNum = LedgerDBPruneZero | LedgerDBPruneNum SecurityParam
```

Reply:
Or `LedgerDBPruneN` or just `LedgerDBPrune`